File: app/src/main/kotlin/com/taleson2wheels/app/data/remote/dto/AuthDtos.kt
package com.taleson2wheels.app.data.remote.dto

import kotlinx.serialization.Serializable

// ── Requests ───────────────────────────────────────────────────────────────

@Serializable
data class LoginRequest(
    val email: String,
    val password: String,
    val deviceId: String? = null,
)

@Serializable
data class RegisterRequest(
    val name: String,
    val email: String,
    val password: String,
    val phone: String? = null,
    val city: String? = null,
    val ridingExperience: String? = null,
    val deviceId: String? = null,
)

@Serializable
data class RefreshRequest(val refreshToken: String)

@Serializable
data class EmailRequest(val email: String)

@Serializable
data class VerifyOtpRequest(val email: String, val code: String)

@Serializable
data class ResetPasswordRequest(
    val email: String,
    val code: String,
    val password: String,
)

// ── Responses ──────────────────────────────────────────────────────────────

@Serializable
data class TokenPair(
    val accessToken: String,
    val refreshToken: String,
    val expiresIn: Int = 900,
)

/** `/auth/login` and `/auth/register` — token pair plus the signed-in user. */
@Serializable
data class AuthSession(
    val accessToken: String,
    val refreshToken: String,
    val expiresIn: Int = 900,
    val user: UserDto,
)

/** `/auth/me` wraps the user in a `{ "user": ... }` object. */
@Serializable
data class MeResponse(val user: UserDto)

File: ios/T2W/Data/Remote/DTO/AuthDTOs.swift
import Foundation

// Shapes mirror the Android `AuthDtos.kt` / T2W `docs/openapi-v1.yaml`.

// MARK: - Requests

struct LoginRequest: Encodable {
    let email: String
    let password: String
    var deviceId: String? = nil
    var platform: String = "ios"
}

struct RefreshRequest: Encodable {
    let refreshToken: String
    var deviceId: String? = nil
    var platform: String = "ios"
}

// MARK: - Responses

/// `/auth/login` and `/auth/register` → token pair + the signed-in user.
struct AuthSuccess: Decodable {
    let accessToken: String
    let refreshToken: String
    /// ISO-8601 instant; the opaque refresh token rotates (~60 days).
    let refreshTokenExpiresAt: String
    let user: UserDTO
}

/// `/auth/refresh` → a fresh token pair (no user payload).
struct RefreshSuccess: Decodable {
    let accessToken: String
    let refreshToken: String
    let refreshTokenExpiresAt: String
}

/// `/auth/me` wraps the user in `{ "user": ... }`.
struct MeResponse: Decodable {
    let user: UserDTO
}
